src/helper/Utils/GetDataFromSnapshot.js

```diff
@@ -0,0 +1,17 @@
+export const GetDataFromSnapshot = async (snapshot) => {
+  const data = [];
+  return await snapshot.onSnapshot(
+    (docs) => {
+      const currentState = [];
+      docs.forEach((doc) => {
+        console.log(doc.data()); // individual doc object
+        currentState.push(doc.data());
+      });
+      data.push(currentState);
+      console.log(currentState); // all the stored docs
+    },
+    (error) => {
+      console.log(error);
+    }
+  );
+};
```

Comment:
This is `utility` to `GetDataFromSnapshot` we will return from our collection, and dive into individual document loop through it and store into an array. And once we will get the all data, then we return back to our reducer fn.